File: src/api/Models/Organization.cs
namespace Trey.Api.Models;

public enum Category
{
  FacultyAndUmbrella,
  Hobby,
  StudentAssociation,
  Other
}

public class Person
{
  public required string Name { get; set; }
  public string? Email { get; set; }
  public string? Phone { get; set; }
  public string? TelegramNick { get; set; }
  public bool? HasBankingAccount { get; set; }
}

public class Facility
{
  public string? Campus { get; set; }
  public string? Building { get; set; }
  public string? RoomCode { get; set; }
  public Person? ContactPerson { get; set; }
  public string? OtherInfo { get; set; }
}

public class TimeRange
{
  public DateTime Start { get; set; }
  public DateTime End { get; set; }
}

public class Organization
{
  public Guid Id { get; set; } = Guid.NewGuid();
  public Category? Category { get; set; }
  public string? Website { get; set; }
  public required string Name { get; set; }
  public string? ShortName { get; set; }
  public int? FoundingYear { get; set; }
  public required TimeRange OperatingPeriod { get; set; }
  public required string Email { get; set; }
  public required Person Chairperson { get; set; }
  public List<Person>? Boardmembers { get; set; }
  public List<Person>? SignatureRightsOwners { get; set; }
  public Person? IntraRightsOwner { get; set; }
  public int MemberCount { get; set; }
  public int? TreyMemberCount { get; set; }
  public string? IBAN { get; set; }
  public List<string>? ReservationRightsEmails { get; set; }
  public Dictionary<string, string[]>? EmailLists { get; set; }
  public Facility? AssociationFacility { get; set; }
}

File: src/web/src/models/organization.ts
export enum Category {
  FacultyAndUmbrella = 0,
  Hobby = 1,
  StudentAssociation = 2,
  Other = 3,
}

export interface Person {
  name: string
  email?: string
  phone?: string
  telegramNick?: string
  hasBankingAccount?: boolean
}

export interface Facility {
  campus?: string
  building?: string
  roomCode?: string
  contactPerson?: Person
  otherInfo?: string
}

export interface TimeRange {
  start?: string
  end?: string
}

export interface Organization {
  id: string
  category?: Category
  website?: string
  name: string
  shortName?: string
  foundingYear?: number
  operatingPeriod: TimeRange
  email: string
  chairperson: Person
  boardmembers?: Person[]
  signatureRightsOwners?: Person[]
  intraRightsOwner?: Person
  memberCount: number
  treyMemberCount?: number
  iban?: string
  reservationRightsEmails?: string[]
  emailLists?: Record<string, string[]>
  associationFacility?: Facility
}
